File: db4oj/core/src/com/db4o/internal/fieldindex/IndexedPathIterator.java
/* Copyright (C) 2006   Versant Inc.   http://www.db4o.com */

package com.db4o.internal.fieldindex;

import com.db4o.foundation.*;
import com.db4o.internal.btree.*;

final class IndexedPathIterator extends CompositeIterator4 {
	
	private IndexedPath _path;
	
	public IndexedPathIterator(IndexedPath path, Iterator4 iterator) {
		super(iterator);
		_path = path;
	}

	protected Iterator4 nextIterator(final Object current) {
		final FieldIndexKey key = (FieldIndexKey) current;
		return _path.search(new Integer(key.parentID())).keys();
	}

}

File: db4o.net/Db4objects.Db4o/Db4objects.Db4o/Internal/Fieldindex/IndexedPathIterator.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using System.Collections;
using Db4objects.Db4o.Foundation;
using Db4objects.Db4o.Internal.Btree;
using Db4objects.Db4o.Internal.Fieldindex;

namespace Db4objects.Db4o.Internal.Fieldindex
{
	internal sealed class IndexedPathIterator : CompositeIterator4
	{
		private IndexedPath _path;

		public IndexedPathIterator(IndexedPath path, IEnumerator iterator) : base(iterator
			)
		{
			_path = path;
		}

		protected override IEnumerator NextIterator(object current)
		{
			IFieldIndexKey key = (IFieldIndexKey)current;
			return _path.Search(key.ParentID()).Keys();
		}
	}
}
